Added in support for Gitbash on Windows OS

jdaniel-cpts opened the pull request:
# Description

Link to Feature Request:  https://github.com/devnw/gvm/issues/30

New Feature: Ability for gvm to support Gitbash running on Windows OS.  (64bit Windows)

1. Added switching to support Windows OS so that script downloads zip file vs tar for alternative OS
2. Updated checksum process to use Windows native checksum for sha256
3. Generalized the downloaded file extension so it's a clean switch between .zip and .tar
4. Since Windows doesn't handle symlinks gracefully had to introduce an explicit delete of the symlink directory before any new link is created.  (Windows only). 


## Examples

Sample Run below of a simple go version switch (all credit to the gvm team, this is simply a port to Windows + Gitbash)

![image](https://user-images.githubusercontent.com/50121968/174152123-3a4cbf3c-815e-4fcb-a5bd-e38a71c4a6d7.png)

In order to test make sure you are running a modern version of Windows i.e. Windows 10 64bit.   

Install Gitbash:  https://www.makeuseof.com/install-git-git-bash-windows/   ( ✔️  Make sure you allow symlinks during install because gvm depends on it to link user to the elected version of go).

jdaniel-cpts committed:
Added in support for Gitbash on Windows OS

jdaniel-cpts commented on gvm:
```diff
@@ -226,7 +233,17 @@ update() {
+          if [[ "$OSTYPE" == "msys"* ]] 
+        then
+            FILESUM=$(certutil -hashfile $0 sha256 | tail -n2 | head -n1)
+            EXPECTEDSUM=$(curl -sL "$SUMPATH" | sed 's/  -//')  # ⚠️ additional 2xspace + dash may also need to be addressed on unix installs
```
This was interesting as I noticed the additional spaces and dash being appended to $EXPECTEDSUM coming from the $SUMPATH request.  I don't have a lot of experience in bash so this may be expected in the Unix environment.  Specifically when using shasum.   Just thought I'd call it out that I had to strip these appended characters in order to accurately compare the file downloaded to the expected checksum.

benjivesterby replied:
I'll look into this

benjivesterby commented:
Thanks for the PR. I'll try and get to it in the next day or so. I'll stand up a VM to test as well.

jdaniel-cpts commented:
> Thanks for the PR. I'll try and get to it in the next day or so. I'll stand up a VM to test as well.

Thank you!  No rush at all.

jdaniel-cpts committed:
Cleaned up a lint issue and also some of my leftover debugging

benjivesterby commented on gvm:
```diff
@@ -2,7 +2,7 @@
 ###############################################################################
 # GENERATED VERSION! DO NOT CHANGE!                                           #
-SCRIPTVERSION="local-dev"
+SCRIPTVERSION="latest"
```
Can you update this to `local-dev`? The script version should be `local-dev` for changes. It's updated by the CI/CD pipeline to match the tagged version prior to generating the sum file for comparison on update.

jdaniel-cpts replied:
No problem.  Just updated in latest commit.

jdaniel-cpts committed:
Update to support CI/CD

Script version should read local-dev for any changes.  This is in support of the CI/CD pipeline.

benjivesterby commented:
@jdaniel-cpts Weekend ended up busier than expected. Will try and get to this this week. Sorry for the delay.

jdaniel-cpts commented:
> @jdaniel-cpts Weekend ended up busier than expected. Will try and get to this this week. Sorry for the delay.

No problem!
